viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -66,90 +69,120 @@ int CartoFacade::AddSensorReading(viam_carto_sensor_reading *sr) {
 }  // namespace carto_facade
 }  // namespace viam
 
-extern int viam_carto_init(const viam_carto **ppVC, const viam_carto_config c,
-                           const viam_carto_algo_config ac, char **errmsg) {
+extern int viam_carto_lib_init(viam_carto_lib **ppVCL) {
+    if (ppVCL == nullptr) {
```

Comment:
Wouldnt we want this to be pointing to nothing originally because we are filling it in?

Reply:
Looked into this more on my own and I think I understand now

Reply:
Not quite:
If `ppVCL` is NULL then line 87 will throw the following exception when we try to dereference it:
```
unknown location(0): fatal error: in "CartoFacadeCPPAPI/CartoFacade_lib_init_terminate": memory access violation at address: 0x00000000: no mapping at fault address
```